### PR TITLE
Broadcast source unit test inval create

### PR DESCRIPTION
**Pull request by Thalley**

### CODE REVIEW

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c [hunk 1]
```diff
@@ -219,3 +219,235 @@ ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_create_start_send
...
+
+ZTEST_F(bap_broadcast_source_test_suite,
+	test_broadcast_source_create_inval_subgroup_params_codec_cfg_cid)
+{
+	struct bt_bap_broadcast_source_create_param *create_param = fixture->create_param;
+	struct bt_bap_broadcast_source_subgroup_param *subgroup_params = &create_param->params[0];
+	struct bt_audio_codec_cfg *codec_cfg = subgroup_params->codec_cfg;
+	int err;
+
+	codec_cfg->id = 0x06;
```
Comment (larsgk) [inline tests/bluetooth/audio/bap_broadcast_source/src/main.c:387]: Maybe use BT_HCI_CODING_FORMAT_LC3 from hci_types (PR: https://github.com/zephyrproject-rtos/zephyr/pull/62525 )

**Review by larsgk — DISMISSED** [review]

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c (hunk 1 above)
Comment (Thalley) [reply tests/bluetooth/audio/bap_broadcast_source/src/main.c:387]: Will update after one of the 2 PRs have been merged :)

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c [hunk 2]
```diff
@@ -268,3 +268,234 @@ ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_start_inval_doubl
 	err = bt_bap_broadcast_source_start(fixture->source, &ext_adv);
 	zassert_not_equal(0, err, "Did not fail with starting already started source");
 }
+ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_create_inval_param_null)
```
Comment (cvinayak) [inline tests/bluetooth/audio/bap_broadcast_source/src/main.c:271]:
```suggestion

ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_create_inval_param_null)
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c (hunk 1 above)
Comment (Thalley) [reply tests/bluetooth/audio/bap_broadcast_source/src/main.c]: Updated

**Review by cvinayak — DISMISSED** [review]

**Review by larsgk — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by carlescufi** [comment]
@cvinayak @larsgk PTAL

**Review by cvinayak — APPROVED** [review]

**Review by larsgk — APPROVED** [review]

**Pull request merged**